doc/man1/openssl-enc.pod.in

```diff
@@ -97,13 +97,19 @@ Base64 process the data. This means that if encryption is taking place
 the data is base64 encoded after encryption. If decryption is set then
 the input data is base64 decoded before being decrypted.
 
+When the B<-A> option not given,
+on encoding a newline is inserted after each 64 characters, and
+on decoding a newline is expected among the first 1024 bytes of input.
```

Comment:
This seems to be documenting some implementation specific thing. I'm not sure that we should document it that way. I think it should just say that it should be split in multiple lines.

Reply:
This sentence gives information relevant to users, regarding the input/output behavior.
It is an abstraction of the info given in `BIO_f_base64.pod`, which also says:
`For writing, by default output is divided to lines of length 64 characters ...`

While the exact number may be an implementation detail in general,
RFC 1421 is very specific about this, whereas later base64 specs are more relaxed on this.
So I think it is good that the user knows what exactly (s)he gets.

Reply:
I think that the first line, and only the first line, being limited to 1024 is really not something we should document. If we document it to just say lines should be shorter than 1024, that's good for me.

Reply:
I currently also don't see why on decoding we can't accept anything, just ignore CR/LF.

Reply:
> I think that the first line, and only the first line, being limited to 1024 is really not something we should document. If we document it to just say lines should be shorter than 1024, that's good for me.

My aim is to somehow document the sad state of affairs (how it apparently has been since long).
Meanwhile I think I got your point: better not document buggy behavior as official behavior.
So I've moved the text in `doc/man3/BIO_f_base64.pod` describing the behavior in case there is no newline among the first 1024 input bytes to the BUGS section.

Reply:
> I currently also don't see why on decoding we can't accept anything, just ignore CR/LF.

Certainly it would be much better if the implementation at least did not pose limits on where to find any required newlines or even just ignored all CR/LF (though the latter is already the case with `BIO_FLAGS_BASE64_NO_NL`).

Yet after having a closer look at the code, I decided not to try fixing the bug(s) myself.

Reply:
I also don't think we should document the oddities, we should fix them :grinning:

Reply:
I'm pretty sure that what I added here to the main body of `openssl-enc.pod.in`:
```
When the B<-A> option not given,
on encoding a newline is inserted after each 64 characters, and
on decoding a newline is expected among the first 1024 bytes of input.
```
was meant to be a feature.
and I can understand that when `BIO_FLAGS_BASE64_NO_NL` is not set that the line lengths are meant to be restricted.
Please also note that is information was already present in the docs, namely in `BIO_f_base64.pod`.
Therefore I propose putting this text in all branches.

OTOH, there are the oddities, and upon the comments by @kroeckx  I made sure that in both doc files they are only mentioned in their BUGS sections:
```
On decoding, if the flag B<BIO_FLAGS_BASE64_NO_NL> is not set and
the first 1024 bytes of input do not include a newline character
the first two lines of input are ignored.
```
I suggest also putting these texts in all branches at first,
and having a follow-up PR for `master` that fixes them and of course removes the respective sentences from the BUGS sections.

Reply:
@t8m, can you agree with what I wrote above?